compiler/resolution/functionResolution.cpp

```diff
@@ -469,15 +471,19 @@ resolveUninsertedCall(BlockStmt* insideBlock,
   INT_ASSERT(block->parentSymbol);
 
   block->insertAtHead(call);
-  resolveCall(call);
+  if (checkonly && !call->primitive) {
+    resolveNormalCall(call, checkonly);
+  } else {
+    resolveCall(call);
```

Comment:
Should we add INT_ASSERT(!checkonly) in this case, i.e. disallow calling this function on a primitive call with checkonly==true ?